packages/components/src/hooks/use-tree-view-keyboard-interaction.ts

```diff
@@ -0,0 +1,208 @@
+import React, { useCallback, useEffect } from 'react';
+import { KeyCodes } from '../common';
+
+interface TreeViewKeyboardInteractionsParams {
+    eventsRoot?: React.RefObject<HTMLElement>;
+    focusedItemId: string | undefined;
+    open: (itemId: string) => void;
+    close: (itemId: string) => void;
+    focus: (itemId: string) => void;
+    select: (itemId: string) => void;
+    isOpen: (itemId: string) => boolean;
+    isEndNode: (itemId: string) => boolean;
+    getPrevious: (itemId: string) => string | undefined;
+    getNext: (itemId: string) => string | undefined;
+    getParent: (itemId: string) => string | undefined;
+    getFirstChild: (itemId: string) => string | undefined;
+    getFirst: () => string | undefined;
+    getLast: () => string | undefined;
+}
+
+interface KeyboardInteractionConfiguration {
+    /**
+     * In a single-select tree, moving focus may optionally unselect the previously selected node
+     * and select the newly focused node.
+     */
+    selectionFollowsFocus?: boolean;
+    /**
+     * Non-standard behavior: when focus is on an end node, pressing the right arrow key will select
+     * the next sibling, or, in case of last child, the next sibling of the parent node.
+     */
+    endNodeExpandSelectsNext?: boolean;
+}
+
+/**
+ * Trying to align to https://www.w3.org/WAI/ARIA/apg/patterns/treeview/ "Keyboard Interaction" section;
+ * for non-standard behavior, see the `KeyboardInteractionConfiguration` configuration options.
+ */
+export const useTreeViewKeyboardInteraction = ({
+    eventsRoot,
+    focusedItemId,
+    isEndNode,
+    getPrevious,
+    getNext,
+    getFirst,
+    getLast,
+    getParent,
+    getFirstChild,
+    isOpen,
+    open,
+    close,
+    focus,
+    select,
+    endNodeExpandSelectsNext = true, // TODO: move to Codux, default should be false
+    selectionFollowsFocus = true, // TODO: move to Codux, default should be false
```

Comment:
I think it should be moved to a "keyboardInteractionOptions" prop on the Tree and passed to this hook, and defaults would be set during the destructuring assignment